### PR TITLE
Remove the hex dependency

### LINKED ISSUE #332: Use `bitcoin_hashes::hex` instead of `crate hex`
This creates problem with defining new hash types, since the `hash_newtype` macro from bitcoin_hash uses it's own hex mod, and rust-bitcoin uses different crate hex.

### PR DESCRIPTION
**Pull request by elichai**
This removes the hex dependecy from the tests.
I had to leave the `hex_hash!` macro because here https://github.com/rust-bitcoin/rust-bitcoin/blob/master/src/util/bip143.rs#L143 I needed to use that instead of `SigHash::from_hex` because the latter requires the hex to be in reverse ordering.

FYI currently even non tests will build `hex` because of `#[cfg(any(test, feature = "serde"))] extern crate hex;` ~This is no longer true as of https://github.com/rust-bitcoin/bitcoin_hashes/pull/71~ (I confued myself. even though the hex crate *isn't used* in non-test code it is still compiled and linked for that reason)

Closes #332

### CODE REVIEW

**Comment by elichai** [comment]
This won't work until bitcoin_hashes#71 will get merged because the `hash_newtype!` currently *require* the hex crate

**Comment by elichai** [comment]
Now that https://github.com/rust-bitcoin/bitcoin_hashes/pull/71 is merged this should is mergeable.

File: src/blockdata/constants.rs [hunk 1]
```diff
@@ -84,7 +84,7 @@ fn bitcoin_genesis_tx() -> Transaction {
 
     // Outputs
     let out_script = script::Builder::new()
-        .push_slice(&Vec::<u8>::from_hex("04678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112de5c384df7ba0b8d578a4c702b6bf11d5f").unwrap())
```
Comment (stevenroose) [inline src/blockdata/constants.rs]: Ah that's really cool :) I guess `Vec<u8>` is the only vec `FromHex` is defined on?

Comment (elichai) [reply src/blockdata/constants.rs]: It wouldn't make much sense on any other Vec :)  (without getting into endianness lol )

**Pull request merged**